app/scripts/shell/dimSearchFilter.directive.js

```diff
@@ -37,9 +55,27 @@
       }, 300);
     });
 
-    vm.filter = function() {
+    $scope.blurFilterInputIfEmpty = function () {
```

Comment:
Set controller properties on 'vm' because that goes on the controller instead of scope.  This keeps scope clean.  

This affects each of the properties on line 58-76.